app/src/main/java/de/grobox/transportr/settings/SettingsManager.kt

```diff
@@ -92,6 +97,26 @@ class SettingsManager @Inject constructor(private val context: Context) {
             }
         }
 
+    val proxy: Proxy
+        get() {
+            if (!settings.getBoolean(PROXY_ENABLE, false))
+                return Proxy.NO_PROXY
+            return try {
+                val typeStr = settings.getString(PROXY_PROTOCOL, null)
+                val type = when (typeStr) {
+                    "SOCKS" -> Proxy.Type.SOCKS
+                    "HTTP" -> Proxy.Type.HTTP
+                    else -> throw IllegalStateException("Illegal proxy type: " + typeStr)
+                }
+                val host = settings.getString(PROXY_HOST, null)
+                val port = Integer.parseInt(settings.getString(PROXY_PORT, null)!!)
+                Proxy(type, InetSocketAddress(InetAddress.getByName(host), port))
```

Comment:
What kind of hosts are accepted here? E.g. do they need `http(s)://`, IP adresses and/or domains?

Reply:
IP addresses and domains, the protocol and port are handled by other inputs. The default is 127.0.0.1.